doc/Programming-guide/Key-platform-features/Devices-provisioning-and-registration/index.md

```diff
@@ -9,77 +9,75 @@ sort_idx: 10
+The resolution response from the Bootstrap service contains a prioritized list of Operations services with their connectivity details and is signed with the Bootstrap service private key.
+The endpoint verifies the signature of the response using the Bootstrap public key available from the build time parameters and saves the received information.
 
-### Endpoint Authorization
+### Endpoint authorization
 
-Endpoint authorization is done by validating endpoint credentials using corresponding credentials service. 
-Credentials service may be configured separately for each application use either Admin UI or 
-[Admin REST API]({{root_url}}Programming-guide/Server-REST-APIs/#!/Application/editApplication). 
+Endpoint authorization is done by validating the endpoint credentials using the corresponding credentials service.
+Credentials service can be configured for each Kaa client individually.
+To do this, you can use the [Administration UI]({{root_url}}Glossary/#administration-ui) or the corresponding [server REST API]({{root_url}}Programming-guide/Server-REST-APIs/#!/Application/editApplication).
 
 ![credential](attach/credentials-for-apps.png)
 
-At the moment of writing Kaa endpoint credentials is identifier is based on hash of endpoint RSA Key Pair.
-During the start of new session, operations service lookup credentials service based on application identifier 
-and uses this credentials service to lookup status of provided credentials object. 
-If Credentials object is not found or this credentials are in use by other endpoint, new session is rejected by Kaa node.
-
-### Endpoint Registration information
+The endpoint credentials ID is generated based on the RSA key pair hash of the endpoint.
+When a new session starts, the Operations service challenges the Credentials service to retrieve credentials status.
+If the credentials are not valid or in use by other endpoint, Kaa node will reject the session.
 
-Kaa platform user is able to provision server-side profile for new endpoints using information about endpoint credentials 
-(See following [Admin REST API]({{root_url}}Programming-guide/Server-REST-APIs/#!/Device_management/provisionCredentials) call for more details). 
-This server-side profile will be used during endpoint registration. 
-Registration service is responsible for storage and queries of this information based on credentials or endpoint identifier.
-This feature may be useful in multiple use-cases. 
-For example, endpoint manufacturer may provision some secure information to the server-side endpoint profile based on security keys that are available only during manufacturing process.
+### Endpoint registration data
 
+Kaa platform user can use the endpoint credentials data to provision server-side profiles for new endpoints.
+This server-side profile will be used for endpoint registration.
```

Comment:
Not "for", but actually "during": server-side EP profile actually gets copied from the storage in credentials service